modules/core/src/shaderlib/lighting/lighting.js

```diff
@@ -58,25 +59,33 @@ function getUniforms(opts = INITIAL_MODULE_OPTIONS) {
 
     lightsPosition = DEFAULT_LIGHTS_POSITION,
     lightsStrength = DEFAULT_LIGHTS_STRENGTH,
-    coordinateSystem = COORDINATE_SYSTEM.LNGLAT,
-    coordinateOrigin = DEFAULT_COORDINATE_ORIGIN,
+    coordinateSystem: fromCoordinateSystem = COORDINATE_SYSTEM.LNGLAT,
+    coordinateOrigin: fromCoordinateOrigin = DEFAULT_COORDINATE_ORIGIN,
     modelMatrix = null,
 
     ambientRatio = DEFAULT_AMBIENT_RATIO,
     diffuseRatio = DEFAULT_DIFFUSE_RATIO,
     specularRatio = DEFAULT_SPECULAR_RATIO
   } = opts.lightSettings;
 
+  let coordinateSystem = opts.coordinateSystem;
+  let coordinateOrigin = opts.coordinateOrigin;
+
+  if (context.project_uCoordinateSystem === PROJECT_COORDINATE_SYSTEM.LNGLAT_AUTO_OFFSET) {
+    coordinateSystem = COORDINATE_SYSTEM.LNGLAT_OFFSETS;
+    coordinateOrigin = context.project_coordinate_origin;
```

Comment:
Who is providing the context?

Reply:
The `context` parameter contains all uniforms generated by a module's dependencies (in this case `project`).